packages/@expo/fingerprint/src/ProjectWorkflow.ts

```diff
@@ -0,0 +1,224 @@
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs/promises';
+import { glob } from 'glob';
+import createIgnore, { Ignore as SingleFileIgnore } from 'ignore';
+import { type Minimatch } from 'minimatch';
+import path from 'path';
+
+import { resolveExpoConfigPluginsPackagePath } from './ExpoResolver';
+import { type Platform, type ProjectWorkflow } from './Fingerprint.types';
+import { isIgnoredPathWithMatchObjects, pathExistsAsync } from './utils/Path';
+
+/**
+ * Replicated project workflow detection logic from eas-cli:
+ * https://github.com/expo/eas-cli/blob/25cc5551899d1ed03a09e04fd5f13a9ad485bd3a/packages/eas-cli/src/project/workflow.ts
+ */
+
+export async function resolveProjectWorkflowAsync(
+  projectRoot: string,
+  platform: Platform,
+  fingerprintIgnorePaths: Minimatch[]
+): Promise<ProjectWorkflow> {
+  const configPluginsPackageRoot = resolveExpoConfigPluginsPackagePath(projectRoot);
+  if (configPluginsPackageRoot == null) {
+    return 'unknown';
+  }
+  const { AndroidConfig, IOSConfig } = require(configPluginsPackageRoot);
```

Comment:
Just curious, why not depend on `@expo/config-plugins`?

Reply:
by design, i want to keep fingerprint to decouple with expo dependencies and sdk versions. that's why i tries to resolve the expo packages from the project rather than having the direct dependency in fingerprint

Reply:
The `expo` package depends on `@expo/fingerprint` which also depends on `@expo/config-plugins`? I guess this dependency tree doesn't imply `@expo/fingerprint` can depend on `@expo/config-plugins`, but to me it feels pretty reliably version-locked?

I guess the issue with resolving this dynamically is that the API for `AndroidConfig` or `IOSConfig` changes (has in the past which has caused issues in the EAS CLI).

Reply:
for `expo -> @expo/fingerprint` chain, that is true. but there are some other use cases without expo, e.g. https://github.com/software-mansion/radon-ide/blob/5e70057092440d41a49461229d45cb664abf3686/packages/vscode-extension/package.json#L732

though `AndroidConfig` / `IOSConfig` changes from time to time, the `AndroidConfig.Paths` / `IOSConfig.Paths` looks quite stable. if it changes frequently in the future, we can implement the `AndroidConfig.Paths.getAndroidManifestAsync` and `IOSConfig.Paths.getPBXProjectPath` in fingerprint anyway.

Reply:
Got it. Should we add some API stability tests to `@expo/config-plugins` like we did for fingerprint (for the same reason, we wanted to require it)? https://github.com/expo/expo/pull/32493

Reply:
good call, adding in the latest commit